### PR TITLE
add a mac build to circleci

### PR DESCRIPTION
**Pull request by borg323**
This PR does several things:
1. Adds mac builds to circleci, generating a universal (x86_64 and arm64) macos binary.
2. Removes clang linux builds (as the mac uses clang) and stops storing ubuntu 18.04 binaries that nobody uses.
3. Fixes non-cross arm builds with ispc.

### CODE REVIEW

File: cross-files/aarch64-darwin [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ld = 'ld'
+
+[built-in options]
+c_args = ['-arch', 'arm64']
+cpp_args = ['-arch', 'arm64']
+objc_args = ['-arch', 'arm64']
+objcpp_args = ['-arch', 'arm64']
+c_link_args = ['-arch', 'arm64']
+cpp_link_args = ['-arch', 'arm64']
+objc_link__args = ['-arch', 'arm64']
```
Comment (almaudoh) [inline cross-files/aarch64-darwin:26]: Seems like the double underscore might be a typo.

Comment (borg323) [reply cross-files/aarch64-darwin:26]: Indeed, so this is not needed - I'll leave it in for future reference.

**Comment by gsobala** [comment]
Meson fails for me as `host_machine.cpu_family() == 'arm'` evaluates as false on Macbook Pro - needs aarch64

See <https://mesonbuild.com/Reference-tables.html#cpu-families>

**Comment by borg323** [comment]
> Meson fails for me as `host_machine.cpu_family() == 'arm'` evaluates as false on Macbook Pro - needs aarch64
> 
> See https://mesonbuild.com/Reference-tables.html#cpu-families
Fixed

**Review by mooskagh — APPROVED** [review]

**Pull request merged**